testsuite/tests/typing-misc/let_rec_approx.ml

```diff
@@ -0,0 +1,23 @@
+(* TEST
+   * expect
+*)
+
+module M = struct type t = A | B end
+let rec f () = g A
+and g (x : M.t) = f ()
+[%%expect{|
+module M : sig type t = A | B end
+val f : unit -> 'a = <fun>
+val g : M.t -> 'a = <fun>
+|}]
+
+let rec f () = g 42
+and g (x : string) = f ()
+[%%expect{|
+Line 1, characters 17-19:
+1 | let rec f () = g 42
+                     ^^
+Error: This expression has type int but an expression was expected of type
+         string
+|}]
+
```

Comment:
Could you include a case of optional argument, to make sure that this is handled correctly?

Reply:
Done